src/components/__tests__/TableWithInlineEditing.tests.js

```diff
@@ -0,0 +1,261 @@
+            item,
+            new MouseEvent('click', {
+                bubbles: true,
+                cancelable: true
+            })
+        );
+        const input = getByDisplayValue('choice2');
+        fireEvent.change(input, {
+            target: { value: 'choice3' }
+        });
+
+        const newContent = [
+            {
+                code: 1,
+                description: 'Descripticon',
+                options: 'choice1'
+            },
+            {
+                code: 22,
+                description: 'Descrip',
+                options: 'choice3'
+            },
+            { code: 333, description: '3rd one', options: 'choice1' },
+            { code: 4, description: 'four', options: 'choice1' }
+        ];
+
+        expect(updateContent).toHaveBeenCalledWith(newContent);
+    });
+
+    test('should update last row description', () => {
+        const { getByText, getByDisplayValue } = render(
+            <TableWithInlineEditing {...defaultProps} />
+        );
+        const item = getByText('four');
+        fireEvent(
+            item,
+            new MouseEvent('click', {
+                bubbles: true,
+                cancelable: true
+            })
+        );
+        const input = getByDisplayValue('four');
+        fireEvent.change(input, {
+            target: { value: 'four4' }
+        });
+
+        const newContent = [
+            {
+                code: 1,
+                description: 'Descripticon',
+                options: 'choice1'
+            },
+            {
+                code: 22,
+                description: 'Descrip',
+                options: 'choice2'
+            },
+            { code: 333, description: '3rd one', options: 'choice1' },
+            { code: 4, description: 'four4', options: 'choice1' }
+        ];
+
+        expect(updateContent).toHaveBeenCalledWith(newContent);
+    });
+});
+
+describe('When not allowed to edit', () => {
+    test('should not update to allow input upon click', () => {
+        const { getByText, queryByDisplayValue, queryByRole, queryByText } = render(
+            <TableWithInlineEditing {...defaultProps} allowedToEdit={false} />
+        );
+        const item = getByText('1');
+        fireEvent(
+            item,
+            new MouseEvent('click', {
+                bubbles: true,
+                cancelable: true
+            })
+        );
+
+        expect(queryByDisplayValue('1')).not.toBeInTheDocument();
+        expect(queryByRole('input')).not.toBeInTheDocument();
+        //expect input not to be there, but span still should be
+        expect(queryByText('1')).toBeInTheDocument();
+    });
+});
+
+//the below was necessary because of the great redux state mutation mystery of 2019
+//which was that the state was being mutated without an action
+// because I had mutated a prop directly.
+//it just takes a copy of the content prop before anything is done and
+//compares it to the content after a change is made.
```

Comment:
I think you can ditch this comment, the describe and test messages do enough to explain the issue under test

Reply:
richard wanted a comment just so we knew why I had added the test looking back, but if it makes enough sense without that's all good I'll ditch it!

Reply:
Yeah. If the describe is clear enough then that's fine.